src/types.ts

```diff
@@ -58,14 +58,17 @@ export type RatelimitResponse = {
   pending: Promise<unknown>;
 };
 
-export type Algorithm<TContext> = (
-  ctx: TContext,
-  identifier: string,
-  rate?: number,
-  opts?: {
-    cache?: EphemeralCache;
-  },
-) => Promise<RatelimitResponse>;
+export type Algorithm<TContext> = () => {
+  limit: (
+    ctx: TContext,
+    identifier: string,
+    rate?: number,
+    opts?: {
+      cache?: EphemeralCache;
+    },
+  ) => Promise<RatelimitResponse>;
+  getRemaining: (ctx: TContext, identifier: string) => Promise<number>;
+};
```

Comment:
Do you think there is a better way of doing this? I feel like these should have their own classes

Reply:
Having classes like how exactly? Will it add too many abstractions to it?

Reply:
> I feel like these should have their own classes

Also it will lead to a breaking change based on how the APIS is currently used. As of now the current pattern works well and we can move the reset API to the existing interface and isolate the reset logic for each window.

So do let me know what you guys think.